### ISSUE TITLE
Support Include after changing query type with Select

### ISSUE THREAD

**Issue by jparish9** [aspnet/EntityFrameworkCore#3565]
Windows 10, VS 2015, IIS Express, SQL Server Express 2014, EF 7.0.0-beta8, ASP.NET 6.0.0-beta8

Using an `.Include` with a `.Select` has no effect (does not populate the child object/collection).

Models:

``` c#
public class Blog
{
        public int BlogId { get; set; }

        public string Status { get; set; }

        public ICollection<Post> Posts { get; set; }

        public virtual BlogMetadata BlogMetadata { get; set; }
}
```

``` c#
public class BlogMetadata
    {
        [Key, ForeignKey("Blog")]
        public int BlogId { get; set; }

        public string Metadata { get; set; }

        public virtual Blog Blog { get; set; }
    }
```

``` c#
public class Post
{
        public int PostId { get; set; }

        public int BlogId { get; set; }

        public string PostText { get; set; }

        public Blog Blog { get; set; }
}
```

FK configuration:

```
            modelBuilder.Entity<Post>().HasOne(p => p.Blog).WithMany(b => b.Posts);
            modelBuilder.Entity<Blog>().HasOne(b => b.BlogMetadata).WithOne(m => m.Blog);
```

Populates `BlogMetadata` child object:

``` c#
List<Blog> blogs = _dbContext.Blogs
                .Where(b => b.Status == "A")
                .Include(b => b.BlogMetadata)
                .ToList();
```

Does not populate `BlogMetadata` child object:

``` c#
List<Blog> blogs = _dbContext.Posts
                .Where(p => p.PostText.Contains("test"))
                .Select(p => p.Blog)
                .Include(b => b.BlogMetadata)
                .ToList();
```

This also happens with child collections (`.Include(b => b.Posts)`) or trying the `.Include` before the `.Select`, e.g.

``` c#
List<Blog> blogs = _dbContext.Posts
                .Where(p => p.PostText.Contains("test"))
                .Include(p => p.Blog.BlogMetadata)
                .Select(p => p.Blog)
                .ToList();
```

or chaining with `.ThenInclude`:

``` c#
List<Blog> blogs = _dbContext.Posts
                .Where(p => p.PostText.Contains("test"))
                .Include(p => p.Blog).ThenInclude(b => b.BlogMetadata)
                .Select(p => p.Blog)
                .ToList();
```

In each case, the generated SQL does not reference `BlogMetadata` at all:

``` sql
SELECT [p.Blog].[BlogId], [p.Blog].[Status]
FROM [Post] AS [p]
INNER JOIN [Blog] AS [p.Blog] ON [p].[BlogId] = [p.Blog].[BlogId]
WHERE [p].[PostText] LIKE ('%' + 'test') + '%'
```

**Comment by AndriySvyryd** [comment]
@smitpatel Also duplicate of #16752?

**Comment by smitpatel** [comment]
This is actually fixed. The last 2 linq query will actually include BlogMetadata on Blog. Earlier we used to ignore it. Fixed by #12737
